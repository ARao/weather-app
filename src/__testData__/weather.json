{
    "last_updated_epoch": 1547153112,
    "last_updated": "2019-01-10 21:45",
    "temp_c": 4,
    "temp_f": 39.2,
    "is_day": 0
}