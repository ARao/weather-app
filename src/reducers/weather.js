import { CURRENT_WEATHER, FORECAST_WEATHER, HISTORY_WEATHER } from '../constants/actionConst';


const initialState = {
  current : {},
  forecast: {},
  histories: [],
};

export default function(state = initialState, action) {
  // console.log(state, action)

  switch (action.type) {
    case CURRENT_WEATHER:
      return {
        ...state,
        current: action.payload
      };
    case FORECAST_WEATHER:
      return {
        ...state,
        forecast: action.payload
      };
    case HISTORY_WEATHER:
      return {
        ...state,
        histories: action.payload
      };
    default:
      return state;
  }
}
