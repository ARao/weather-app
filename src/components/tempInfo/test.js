import React from 'react';
import TempInfo from './index'
import { shallow } from 'enzyme'
import weather from '../../__testData__/weather.json'


describe('HistoryTable', () => {
  it('++++ render Temp info', () => {
    let wrapper = shallow(<TempInfo weather={weather} />);
    expect(wrapper.length).toEqual(1);
  });

  it('++++ snapshot match', () => {
    let wrapper = shallow(<TempInfo weather={weather} />);
    expect(wrapper).toMatchSnapshot();
  });
});
