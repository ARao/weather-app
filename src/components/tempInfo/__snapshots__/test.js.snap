// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTable ++++ snapshot match 1`] = `
<Fragment>
  <span
    className="p-2"
  >
    4
     
    <sup>
      o
    </sup>
    C
  </span>
  <span
    className="p-2"
  >
    2019-01-10 21:45
  </span>
</Fragment>
`;
